piku_dashboard/main.py

```diff
@@ -51,7 +62,7 @@ def home():
         flash(f"Could Not fetch app list: {e}", "error")
         apps = []
 
-    return render_template("home.html", apps=apps, host_info=host_info, self_app=self_app)
+    return render_template("home.html", apps=apps, host_info=host_info, self_app=self_app, github_sha=get_github_sha())
```

Comment:
I'm concerned about hitting the get_github_sha on every page load, could we cache this? 

you could either implement this with a ttl variable or I think i saw a version of functools' lru_cache decorator with a TTL:
https://realpython.com/lru-cache-python/#evicting-cache-entries-based-on-both-time-and-space